ui/src/utils/QualityControl.js

```diff
@@ -40,11 +40,45 @@ export class QualityControl {
             errorMessage: '',
         };
 
+        const audioResult = this.soundCheck();
+
         if (this.audioBuffer.duration < 2.0) {
             qualityResult.success = false;
             qualityResult.errorMessage += 'Audio recording failed: recording was too short. Try again';
+        } else if (audioResult) {
+            qualityResult.success = false;
+            qualityResult.errorMessage += audioResult;
```

Comment:
Because it is in an if statement with else ifs, then if one of those conditions is true, it will not check for the other conditions. So this code only returns one error message.

Reply:
So true. My bad for misreading.